src/libraries/System.Collections/tests/Generic/Queue/Queue.Generic.Tests.cs

```diff
@@ -289,6 +289,53 @@ public void Queue_Generic_TryDequeue_AllElements(int count)
             }
         }
 
+        [Fact]
+        public void Queue_Generic_IterateLastIndexOfMaxSizedQueue_DoesNotOverflow()
+        {
+            Queue<byte> queue;
+            try
+            {
+                queue = new Queue<byte>(Array.MaxLength);
+            }
+            catch (OutOfMemoryException)
+            {
+                // just skip when ctor throws OOM
+                return;
+            }
+
+            // once the internal index is moved (via enqueue/dequeue operations), enumerating
+            // the queue of size up to this value is safe from hitting the corner case.
+            int safeValue = int.MaxValue - Array.MaxLength + 1; // 56 + 1
+
+            // corner case value is any number higher than the safe value.
+            int expectedValue = safeValue + 1; // 58
+
+            // enqueue and dequeue to advance internal head and index
+            // to reach the sweet spot; one less than the max length.
+            for (int i = 0; i < Array.MaxLength - 1; i++)
+            {
+                queue.Enqueue(0);
+                queue.Dequeue();
+            }
```

Comment:
How long does this take to run?

Reply:
This test runs in 10-12 seconds on M1 machine.

Reply:
We can move it to outerloop if it's too slow (in the CI, it will be bit slow than 12 seconds).

Reply:
Yes:
https://github.com/dotnet/runtime/pull/78600#discussion_r1027368770

:-)

Reply:
Ah, I was focusing on [ConditionalFact], forgot that ed1ee5d also added [Outerloop]. :)